src/halmos/sevm.py

```diff
@@ -1960,20 +2019,21 @@ def jump(self, ex: Exec, stack: List[Tuple[Exec, int]], step_id: int) -> None:
                 target_reachable = simplify(dst == target)
                 if ex.check(target_reachable) != unsat:  # jump
                     if self.options.get("debug"):
-                        print(f"We can jump to {target} with model {ex.solver.model()}")
+                        print(
+                            f"We can jump to {target} with model {ex.path.solver.model()}"
+                        )
                     new_ex = self.create_branch(ex, target_reachable, target)
                     stack.append((new_ex, step_id))
         else:
             raise NotConcreteError(f"symbolic JUMP target: {dst}")
 
     def create_branch(self, ex: Exec, cond: BitVecRef, target: int) -> Exec:
-        new_solver = create_solver(
-            timeout=self.options["timeout"], max_memory=self.options["max_memory"]
-        )
-        new_solver.add(ex.solver.assertions())
-        new_solver.add(cond)
-        new_path = ex.path.copy()
-        new_path.append(str(cond))
+        new_path = deepcopy(ex.path)
+        new_path.pending.append(cond)
```

Comment:
style: I had to hunt for this, I couldn't immediately see in Path where things were added to `pending`. Would be mildly cleaner/easier with an `add_pending(cond)` method IMO